extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java

```diff
@@ -186,16 +190,14 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
       // if the response is "unauthorized", refresh the token and try the request again
       if (e.getStatusCode() == 401) {
         monitor.info(() -> "Attempting to refresh authorization token");
-        if (credential.refreshToken()) {
-          // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
-          monitor.info(() -> "Refreshed authorization token successfuly");
-          response = requestFactory
-              .buildPostRequest(new GenericUrl(url + "?" + generateParamsString(parameters)),
-                  httpContent).execute();
-          
-        } else {
-          throw new IOException("Couldn't refresh authorization token after retrying");
-        }
+        // if the credential refresh failed, let the error bubble up via the IOException that gets thrown
+        credential = credentialFactory.refreshCredential(credential);
+        monitor.info(() -> "Refreshed authorization token successfuly");
+
+        // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
```

Comment:
Is this comment still right?

Reply:
Yup - the second post request attempt will error out and we wont try to catch that error